File: Semester 4/Systems for Desing and Implementation/Lab5/Frontend/BookAPI-frontend/src/models/Book.ts
import { BookAuthor } from "./BookAuthor";
import { Genre } from "./Genre";

export interface Book{
    id: number;
    title: string;
    description: string;
    year: number;
    pages: number;
    price: number;
    genreId: number;
    genre: Genre;
    bookAuthors: BookAuthor[];
}

File: Semester 4/Systems for Desing and Implementation/Lab6/Backend/Lab3BookAPI/Model/Book.cs
namespace Lab3BookAPI.Model
{
    public class Book
    {
        public int Id { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public int Year { get; set; }
        public int Pages { get; set; }
        public float Price { get; set; }

        public string? Transcript { get; set; } 

        public int GenreId { get; set; }
        public virtual Genre Genre { get; set; } = null!;


        public virtual ICollection<BookAuthor> BookAuthors { get; set;  }
    }
}